### PR TITLE
Feature/tf 361 allow query of state

### PR DESCRIPTION
**Pull request by GabrielHomewoodUL**
- [x] Test plan updated (by PR creator) 
- [x] Code Review 
- [x] Developer Testing: Key item tests 
- [x] Developer Testing: Execution of the updated test plan (changed items for this work only) 
- [ ] Product Owner Review (Optional) 
- [ ] Update Release Notes on the JIRA ticket for harvest at release time

### CODE REVIEW

File: TF_Service_dotNet/TouchFree_Service/Connection/ClientConnection.cs [hunk 1]
```diff
@@ -121,16 +141,16 @@ private Compatibility GetVersionCompability(string _clientVersion, Version _core
 
             else if (clientVersionParsed.Minor < _coreVersion.Minor)
             {
-                return Compatibility.CLIENT_OUTDATED;
+                return Compatibility.CLIENT_OUTDATED_WARNING;
```
Comment (NatGreenwood-UL) [inline TF_Service_dotNet/TouchFree_Service/Connection/ClientConnection.cs:144]: Why are we warning incompatibility with Minor versions now? Historically we've said that Minor changes in the API are breaking and only Hotfix changes were compatible?

**Review by NatGreenwood-UL — COMMENTED** [review]
Just one comment on the code, going to test this now 👍

File: TF_Service_dotNet/TouchFree_Service/Connection/TouchFreeServiceTypes.cs [hunk 2]
```diff
@@ -25,7 +28,28 @@ internal enum Compatibility
     {
         COMPATIBLE,
         SERVICE_OUTDATED,
-        CLIENT_OUTDATED
+        CLIENT_OUTDATED,
+        SERVICE_OUTDATED_WARNING,
+        CLIENT_OUTDATED_WARNING
+    }
+
+    public enum CameraState
```
Comment (NatGreenwood-UL) [inline TF_Service_dotNet/TouchFree_Service/Connection/TouchFreeServiceTypes.cs]: Was thinking about this, currently if the tracking Service isn't connected, we'll report Camera NOT_CONNECTED even if it is physically, as we can't get at the information about the hardware without the Service. Was going to suggest a third entry here along the lines of "NO_SERVICE" to indicate that but then I realised at that point we may as well combine this into one enum, as we can only (reliably) report 

* Everything fine
* No Camera
* No Service

Do you think it's worth the consolidation here as the scenario of "No Service, Camera" will never actually be possible?

Comment (GabrielHomewoodUL) [reply TF_Service_dotNet/TouchFree_Service/Connection/TouchFreeServiceTypes.cs]: I think that makes sense. I'll start making the changes now

File: TF_Tooling_Web/src/TouchFreeToolingTypes.ts [hunk 3]
```diff
@@ -104,19 +104,13 @@ export enum InteractionType {
...
-// CONNECTED - There is a camera connected
-export enum CameraState {
-    NOT_CONNECTED,
-    CONNECTED,
-}
-
 // Enum: TrackingServiceState
-// UNAVAILABLE - The TouchFree services is not connected to the tracking service
+// UNAVAILABLE - The TouchFree service is not connected to the tracking service
+// NOT_CONNECTED - The TouchFree service is connected to the tracking service but there is not a camera connected
```
Comment (MattGrayUL) [inline TF_Tooling_Web/src/TouchFreeToolingTypes.ts]: this should be NO_CAMERA :)

Comment (GabrielHomewoodUL) [reply TF_Tooling_Web/src/TouchFreeToolingTypes.ts]: Done!

File: TF_Service_dotNet/TouchFree_Service/Connection/TouchFreeServiceTypes.cs (hunk 2 above)
Comment (GabrielHomewoodUL) [reply TF_Service_dotNet/TouchFree_Service/Connection/TouchFreeServiceTypes.cs]: Done!

File: TF_Service_dotNet/TouchFree_Service/Connection/ClientConnection.cs (hunk 1 above)
Comment (GabrielHomewoodUL) [reply TF_Service_dotNet/TouchFree_Service/Connection/ClientConnection.cs:144]: We have come to the decision to allow the service to be minor versions ahead and the client to be hotfix versions ahead. I've pushed up the change for this

**Review by MattGrayUL — DISMISSED** [review]

**Review by MattGrayUL — APPROVED** [review]

**Pull request merged**